primitives/src/analytics.rs

```diff
@@ -76,10 +77,72 @@ pub struct AnalyticsQuery {
     #[serde(default = "default_event_type")]
     pub event_type: String,
     #[serde(default = "default_metric")]
-    pub metric: String,
+    pub metric: Metric,
     #[serde(default = "default_timeframe")]
-    pub timeframe: String,
-    pub segment_by_channel: Option<String>,
+    pub timeframe: Timeframe,
+    pub segment_by: Option<String>,
+    pub start: Option<DateTime<Utc>>,
+    pub end: Option<DateTime<Utc>>,
```

Comment:
We should probably deserialize directly into DateHour.

Also, the `Joi` package usually supports both date (String) & milliseconds (number) but we have to double check if we are going to just leave it to milliseconds.
We use milliseconds almost everywhere in the JSON representaiton.

for more details check:
https://joi.dev/api/?v=17.5.0#date